### PR TITLE
**Breaking:** Refactor Tree Component

### PR DESCRIPTION
**Pull request by TejasQ**
# Summary
This PR refactors the `Tree` component to:
- Be more performant by correctly rendering sections of itself instead of _the whole tree_.
- Use tested drag/drop reorder that allows keyboard/touchscreen/other a11y improvements.

optim

# To be tested

Me
- [ ] No error or warning in the console on `localhost:6060`

Tester 1

- [ ] Things look good on the demo.
  

Tester 2

- [ ] Things look good on the demo.

### CODE REVIEW

File: src/types.ts [hunk 1]
```diff
@@ -10,11 +10,11 @@ export interface DefaultProps {
...
   draggable?: boolean
-  onDragStart?: (ev: React.DragEvent<HTMLElement>) => void
-  onDragEnd?: (ev: React.DragEvent<HTMLElement>) => void
-  onDragEnter?: (ev: React.DragEvent<HTMLElement>) => void
-  onDragExit?: (ev: React.DragEvent<HTMLElement>) => void
-  onDragLeave?: (ev: React.DragEvent<HTMLElement>) => void
-  onDragOver?: (ev: React.DragEvent<HTMLElement>) => void
-  onDrop?: (ev: React.DragEvent<HTMLElement>) => void
+  onDragStart?: (ev: React.DragEvent<HTMLDivElement>) => void
+  onDragEnd?: (event: React.DragEvent<HTMLDivElement>) => void
```
Comment (fabien0102) [inline src/types.ts]: 🤔

Comment (TejasQ) [reply src/types.ts]:
```suggestion
  onDragEnd?: (ev: React.DragEvent<HTMLDivElement>) => void
```

File: src/utils/constants.ts [hunk 2]
```diff
@@ -150,6 +150,9 @@ const font = {
 
     /** 12px */
     fineprint: 12,
+
+    /** 12px */
```
Comment (stereobooster) [inline src/utils/constants.ts]: 11px

File: src/Tree/styledComponents.ts [hunk 3]
```diff
@@ -0,0 +1,60 @@
+import Icon from "../Icon/Icon"
+import { darken } from "../utils"
+import styled from "../utils/styled"
+
+const highlightColor = "#fff26666"
```
Comment (stereobooster) [inline src/Tree/styledComponents.ts]: not in theme?

Comment (TejasQ) [reply src/Tree/styledComponents.ts]: This is a great point 🚀

**Comment by fabien0102** [comment]
@TejasQ The best valentine gift ever 😅

**Comment by fabien0102** [comment]
@TejasQ Some spacing are a bit strange

image

It's not always centered 😕

**Comment by fabien0102** [comment]
Also, if I change the `iniatiallyOpen` on the fly, the tree is still close, I will add an example in the doc to isolate this behaviour ;)

File: src/Tree/README.md [hunk 4]
```diff
@@ -54,47 +50,88 @@ The tree component renders a tree structure with collapsable nodes in a filetree
...
-Tree items can be reordered using drag-and-drop. The API for this feature is low-level and assumes that the tree nodes are generated from data available in a different shape in state, making it dangerous and error-prone to update based on a readily reordered set of nodes.
+```jsx
+<Tree
+  trees={[
+    {
+      label: "Store",
+      highlight: true,
+      childNodes: [
+        {
+          label: "Region",
```
Comment (stereobooster) [inline src/Tree/README.md]: @fabien0102 needs `meta?: Object` in nodes and `onNodeClick: (TreeNode) => void`

File: src/Tree/Tree.tsx [hunk 5]
```diff
@@ -1,339 +1,74 @@
...
+  label: string
+  highlight?: boolean
+  childNodes?: Tree[]
+  initiallyOpen?: boolean
+  tag?: string
+  disabled?: boolean
+  color?: string
+  onClick?: (node: Tree) => void
+  onRemove?: (event: React.MouseEvent<HTMLDivElement, MouseEvent>) => void
+  innerRef?: (element?: HTMLElement | null | undefined) => any
```
Comment (stereobooster) [inline src/Tree/Tree.tsx]:
```suggestion
  innerRef?: (element?: HTMLElement | null) => any
```

File: src/Tree/Tree.tsx [hunk 6]
```diff
@@ -1,339 +1,74 @@
...
-    }))
-  }
-
-  private setReorderSource = (reorderSource?: number[]) => {
-    this.setState(() => ({
-      reorderSource,
-    }))
-  }
+const Tree: React.SFC<TreeProps> = ({ trees, id, draggable }) => {
+  const isLowestLevel = trees.filter(tree => !tree.childNodes || !tree.childNodes.length).length
```
Comment (stereobooster) [inline src/Tree/Tree.tsx]:
```suggestion
  const isLowestLevel = trees.some(tree => !tree.childNodes || !tree.childNodes.length)
```

**Pull request merged**